network.go

```diff
@@ -237,6 +270,89 @@ func getLinkByName(netHandle *netlink.Handle, name string, expectedLink netlink.
 	return nil, fmt.Errorf("Incorrect link type %s, expecting %s", link.Type(), expectedLink.Type())
 }
 
+func connectVMNetwork(netPair *NetworkInterfacePair) error {
+	switch DefaultNetInterworkingModel {
+	case ModelBridged:
+		return bridgeNetworkPair(netPair)
+	case ModelMacVtap:
+		return tapNetworkPair(netPair)
+	case ModelEnlightened:
+		return fmt.Errorf("Unsupported networking model")
```

Comment:
Can you add `DefaultNetInterworkingModel` to the error here and in the error below. Also, due to the similarity of this function and `disconnectVMNetwork`, I wonder if it would be worth making both these functions call a new function that both validates `netPair` and either calls the "connect" or "disconnect" function:

```
func toggleVMNetwork(netPair *NetworkInterfacePair, connect bool) error
```

In fact, with the above function, we wouldn't really need `connectVMNetwork` and `disconnectVMNetwork` as the intent would be clear from the `connect` param.

Reply:
@jodh-intel addressed. 

@amshinde we need to make this even more generic to include SRIOV, VFIO and vhost-user. So having an pair abstraction is not ideal. We need a more generic abstraction that support all models.  I will rework this based on VFIO and vhost-user needs